### ISSUE TITLE
Level-k implementation discrepancy

### ISSUE THREAD

**Issue by yasuiniko** [jrwright/bogota#3]
## TLDR;
Suppose you have two actions and `epsilon = 0.1`. Action 0 is the BR. Then the code would give:
`l1_prediction = [0.95, 0.05]`
and the pseudocode in WLB17 would give: 
`l1_prediction = [0.1, 0.9]`

## Summary: 
The level-k code in the repo is different from the pseudocode in "Predicting human behavior in unrepeated, simultaneous-move games" (WLB17). I was not able to find a canonical implementation of Level-k, as the references from WLB17 ([CGCB01](https://econweb.ucsd.edu/~vcrawfor/CGCrBr01EMT.pdf), [N95](https://www.jstor.org/stable/pdf/2950991.pdf)) do not have any epsilon and seem to play noiseless best-response actions or logit best response actions. 

Here is the relevant part of WLB17:
![](https://user-images.githubusercontent.com/7184841/219155970-3a14b0ee-356f-42ae-98da-27ca827b8099.png)

And here is the relevant code from this repo:
https://github.com/jrwright/bogota/blob/be8f1697f78b8f03d7e30b3945f223479bbcc5a6/bogota/cognitive_hierarchy.py#L133-L148
